src/paging.rs

```
@@ -0,0 +1,8 @@
+/// General paging related code
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum PagetableError {
+	#[error("The accessed virtual address is not mapped")]
+	InvalidAddress,
```

Nit: According to the description, would `NotMapped` be more fitting?

---

Ah, this is just moved.

---

Let's ignore this error until it actually occurs somewhere, or we differentiate between these cases.